ra/ra_test.go

```diff
@@ -77,6 +78,24 @@ func randomDomain() string {
 	return fmt.Sprintf("%x.example.com", bytes[:])
 }
 
+// randomIPv6 creates a random IPv6 netip.Addr for testing. It uses a real IPv6
+// address range, not a test/documentation range.
+//
+// panics if crypto/rand.Rand.Read or netip.AddrFromSlice fails.
+func randomIPv6() netip.Addr {
+	var ipBytes [11]byte
+	_, err := rand.Read(ipBytes[:])
+	if err != nil {
+		panic(err)
+	}
+	ipPrefix, _ := hex.DecodeString("2602080a600")
```

Comment:
hex.DecodeString needs an even-length string, here you're actually swallowing an error that reads `encoding/hex: odd length hex string`.

Instead, I believe you can do this:

```suggestion
	ipPrefix, err := hex.DecodeString("2602080a60")
	if err != nil {
		panic(err)
	}
```